### PR TITLE
Bump to Python 3.8

### PR DESCRIPTION
**Pull request by dellis23**
Progress on #12835

### CODE REVIEW

**Comment by dellis23** [comment]
Related: https://github.com/openxla/iree/issues/12835

**Review by stellaraccident — APPROVED** [review]

**Review by GMNGeoffrey — COMMENTED** [review]
Please also update the other places we document Python requirements:

e.g. most of the hits here: https://github.com/search?q=3.7+repo%3Aopenxla%2Firee&type=code

**Comment by stellaraccident** [comment]
> Please also update the other places we document Python requirements:
> 
> e.g. most of the hits here: https://github.com/search?q=3.7+repo%3Aopenxla%2Firee&type=code

He's out for the night. If this passes, I'll follow up with more surgery.

**Comment by stellaraccident** [comment]
I'm going to go ahead and land this so that Dan is unblocked in his morning. I'll follow up later this evening to handle the rest.

File: build_tools/docker/dockerfiles/base.Dockerfile [hunk 1]
```diff
@@ -69,7 +69,7 @@ RUN ./install_bazel.sh && rm -rf /install-bazel
 WORKDIR /install-python
 
 # Minimum supported Python version
-ARG PYTHON_VERSION=3.7
+ARG PYTHON_VERSION=3.8
```
Comment (ScottTodd) [inline build_tools/docker/dockerfiles/base.Dockerfile:72]: You should rebuild all images that depend on base, not just the "frontends" images after this change. See https://github.com/openxla/iree/tree/main/build_tools/docker#dependencies-between-images

Comment (GMNGeoffrey) [reply build_tools/docker/dockerfiles/base.Dockerfile:72]: Scott is correct. I don't believe this has been fixed afterwards either and our Docker images are in a weird unsynchronized state.

Comment (ScottTodd) [reply build_tools/docker/dockerfiles/base.Dockerfile:72]: (Fixed in https://github.com/openxla/iree/pull/12857)

Comment (GMNGeoffrey) [reply build_tools/docker/dockerfiles/base.Dockerfile:72]: I am rebuilding everything now

Comment (GMNGeoffrey) [reply build_tools/docker/dockerfiles/base.Dockerfile:72]: Ah thanks Scott. I saw that in history and thought it was this PR

Comment (stellaraccident) [reply build_tools/docker/dockerfiles/base.Dockerfile:72]: Thank you Geoffrey - I was just going to say that I could start a rebuild.

**Pull request merged**